### PR TITLE
VACMS-19544: Adding Connect with Us Content Block

### PR DESCRIPTION
**Pull request by davidmpickett**
## Description

Relates to #19544

## Testing done

Manual in tugboat 

## Screenshots

<details><summary>New Connect with Us content block interface</summary>

screencapture-pr19770-drgsfxyuphduaugv2lfadosblcnwadfr-ci-cms-va-gov-block-add-connect-with-us-2024-11-05-15_19_58

</details>
<details><summary>Current Office node Connect with Us interface</summary>

screencapture-prod-cms-va-gov-node-add-office-2024-11-05-19_18_41

</details>
<details><summary>CLP Node view with old way showing VERY OBVIOUS DRAFT CONTENT while new way does not </summary>

Screenshot 2024-11-05 at 7 07 52 PM

</details>
<details><summary>BHLP Node view with old way showing VERY OBVIOUS DRAFT CONTENT while new way does not </summary>

Screenshot 2024-11-05 at 7 10 24 PM

</details>

### Part 1 - Create a new Connect with Us Block
1. Log into the [Tugboat](https://pr19770-drgsfxyuphduaugv2lfadosblcnwadfr.ci.cms.va.gov/) as a Content Admin
2. Go to Content > Blocks > And content block > [Connect with us](https://pr19770-drgsfxyuphduaugv2lfadosblcnwadfr.ci.cms.va.gov/block/add/connect_with_us)
3. Fill out the fields
   - You can copy and paste from the Connect With Us section of these 4 Office nodes
   - [NCA](https://pr19770-drgsfxyuphduaugv2lfadosblcnwadfr.ci.cms.va.gov/node/38441/edit)
   - [VACO](https://pr19770-drgsfxyuphduaugv2lfadosblcnwadfr.ci.cms.va.gov/node/38442/edit)
   - [VBA](https://pr19770-drgsfxyuphduaugv2lfadosblcnwadfr.ci.cms.va.gov/node/38439/edit)
   - [VHA](https://pr19770-drgsfxyuphduaugv2lfadosblcnwadfr.ci.cms.va.gov/node/38440/edit)
5. Save as published 

### Part 2 - Stress test the Connect with Us Block form
1. Log into the [Tugboat](https://pr19770-drgsfxyuphduaugv2lfadosblcnwadfr.ci.cms.va.gov/) as a Content Admin
2. Go to Content > Blocks > And content block > [Connect with us](https://pr19770-drgsfxyuphduaugv2lfadosblcnwadfr.ci.cms.va.gov/block/add/connect_with_us)
3. Attempt to save without entering information in any field
4. Confirm appropriate validation errors occur
5. Attempt to put invalid URLs in the URL field
6. Notice that the Link Text fields become required once something is entered in URL field
7. And vice versa with URL fields being required once text is entered in Link Text 
8. Save it with every different moderation state

### Part 3 - Compare it with the existing Office node form
1. Log into the [Tugboat](https://pr19770-drgsfxyuphduaugv2lfadosblcnwadfr.ci.cms.va.gov/) as a Content Admin
2. Go to Content > Add content > [Office](https://pr19770-drgsfxyuphduaugv2lfadosblcnwadfr.ci.cms.va.gov/node/add/office)
3. Open the Connect with Us panel
4. Confirm that all the fields here are the same as the Connect with Us content block

### Part 4 - Add it to a Benefit Hub Landing Page
1. Log into the [Tugboat](https://pr19770-drgsfxyuphduaugv2lfadosblcnwadfr.ci.cms.va.gov/) as a Content Admin
2. Go to Content and Filter by [Content Type = Benefit Hub Landing Page](https://pr19770-drgsfxyuphduaugv2lfadosblcnwadfr.ci.cms.va.gov/admin/content?title=&type=landing_page&moderation_state=All&owner=All)
3. Choose a BHLP and edit it
4. In the right side bar, expand the Right Rail accordion
5. Scroll downnnn
6. Notice the `Add a "Connect with us" block` and `Connect with us` fields.
7. The first field is the old field - select an option
8. The second field is the new field - select a Connect with Us block you created in Step 1 or 2
9. Save as Published

Screenshot 2024-11-05 at 5 15 44 PM

### Part 5 - Add it to a Campaign Landing Page
1. Log into the [Tugboat](https://pr19770-drgsfxyuphduaugv2lfadosblcnwadfr.ci.cms.va.gov/) as a Content Admin
2. Go to Content and Filter by [Content Type = Campaign Landing Page](https://pr19770-drgsfxyuphduaugv2lfadosblcnwadfr.ci.cms.va.gov/admin/content?title=&type=campaign_landing_page&moderation_state=All&owner=All)
3. Choose a CLP and edit it
4. Scroll down to the Connect with Us accordion and open it
5.  Notice the `Add a "Connect with us" panel` and `Connect with us` fields.
6. The first field is the old field - select an option
7. The second field is the new field - select a Connect with Us block you created in Step 1 or 2
8. Save as Published

### Part 6 - Stress test the BHLP and CLP connection
1. Confirm that the Connect With Us fields works correctly on Node Add
   - [x] https://pr19770-drgsfxyuphduaugv2lfadosblcnwadfr.ci.cms.va.gov/node/add/campaign_landing_page
   - [x] https://pr19770-drgsfxyuphduaugv2lfadosblcnwadfr.ci.cms.va.gov/node/add/landing_page
2. Remove a Connect with us from a BHLP and CLP (set to none)
3. Save it with every different moderation state
4. Look at how stuff looks on Node View (I didn't make the new version as pretty as the old version)

### Part 7 - See the different in published vs unpublished changes
1. Go to the [Blocks list](https://pr19770-drgsfxyuphduaugv2lfadosblcnwadfr.ci.cms.va.gov/admin/content/block) and edit the block you created in Step 1 and attached to a CLP and BHLP in Steps 4 & 5
2. Make some VERY OBVIOUS EDITS
3. Save this revision as DRAFT
4. Go to Content and [Filter by Office](https://pr19770-drgsfxyuphduaugv2lfadosblcnwadfr.ci.cms.va.gov/admin/content?title=&type=office&moderation_state=All&owner=All)
5. Select one of the four Offices with Connect with Us content (the same one you associated with a BHLP and CLP in Steps 4 & 5) and edit it
6. Make some VERY OBVIOUS EDITS
7. Save this revision as DRAFT
8. Go back to the BHLP and CLP nodes that are associated with this Block and Office
9. Notice on Node View that the old Connect With Us is showing the VERY OBVIOUS EDITS while the new Connect With Us is not, because it has a previous published revision
11. Party on, Garth

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.
- [ ] If there are field changes, front end output has been thoroughly checked.

### CODE REVIEW

File: config/sync/workbench_access.access_scheme.section.yml [hunk 1]
```diff
@@ -90,6 +91,10 @@ scheme_settings:
       entity_type: block_content
       bundle: benefit_promo
       field: field_administration
+    -
```
Comment (jv-agile6) [inline config/sync/workbench_access.access_scheme.section.yml:94]: Why the dead space here?

File: config/sync/workbench_access.access_scheme.section.yml [hunk 2]
```diff
@@ -90,6 +91,10 @@ scheme_settings:
...
+      entity_type: block_content
+      bundle: connect_with_us
+      field: field_administration
     -
```
Comment (jv-agile6) [inline config/sync/workbench_access.access_scheme.section.yml:98]: and here

**Review by jv-agile6 — APPROVED** [review]
Followed QA steps and completed as expected with no defects. Reviewed configuration and made comment on config/sync/workbench_access.access_scheme.section.yml line spacing on 94 & line 98. Approval granted

File: config/sync/workbench_access.access_scheme.section.yml (hunk 1 above)
Comment (davidmpickett) [reply config/sync/workbench_access.access_scheme.section.yml:94]: If you view the full file you can see that this is the way the whole thing is formatted. I couldn't tell you if it has a functional purpose, or was just for readability. Following the principle of Chesterton Fence, I just kept with existing pattern

**Review by dsasser — COMMENTED** [review]
I see failing va/tests/phpunit (phpunit) tests:

Screenshot 2024-11-08 at 9 56 35 AM

I can't see what failed but based on what changed I suspect it is [this](https://github.com/department-of-veterans-affairs/va.gov-cms/blob/main/tests/phpunit/Security/RolesPermissionsTest.php) test that failed.

**Comment by laflannery** [comment]
Can we have the new one still say "Add a "Connect with us" block"? It's action oriented and a bit clearer as to what folks should do when they hit this dropdown. Or is there a technical reason why we can't have that any more?
Screenshot 2024-11-08 at 3 21 11 PM

**Comment by davidmpickett** [comment]
> I see failing va/tests/phpunit (phpunit) tests:
> 
> Screenshot 2024-11-08 at 9 56 35 AM
> I can't see what failed but based on what changed I suspect it is [this](https://github.com/department-of-veterans-affairs/va.gov-cms/blob/main/tests/phpunit/Security/RolesPermissionsTest.php) test that failed.

I updated the test and now phpunit tests are passing. Thanks!

**Comment by davidmpickett** [comment]
> Can we have the new one still say "Add a "Connect with us" block"? It's action oriented and a bit clearer as to what folks should do when they hit this dropdown. Or is there a technical reason why we can't have that any more? Screenshot 2024-11-08 at 3 21 11 PM

Updated this on both BHLPs and CLPs. CLPs previously called it a panel, but block is more accurate

Screenshot 2024-11-08 at 5 28 05 PM

Screenshot 2024-11-08 at 5 27 56 PM

**Comment by thejordanwood** [comment]
@davidmpickett I've tested and it seems that everything is working as expected. Did you plan to leave the old label as "Add 'Connect with us' panel" on CLPs or were you going to change that to "block" as well? I could go either way on this.

**Comment by davidmpickett** [comment]
> @davidmpickett I've tested and it seems that everything is working as expected. Did you plan to leave the old label as "Add 'Connect with us' panel" on CLPs or were you going to change that to "block" as well? I could go either way on this.

That label will be hidden in a follow up ticket, so I just left it as is for now.

**Review by jv-agile6 — APPROVED** [review]
Reviewed code changes to field.field.node.campaign_landing_page.field_connect_with_us.yml and field.field.node.landing_page.field_connect_with_us.yml. No defects found. Approval granted

**Pull request merged**